shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1168,11 +1169,19 @@ - (void)setTextInputClient:(int)client withConfiguration:(NSDictionary*)configur
       break;
   }
 
-  // Clean up views that should no longer be in the view hierarchy according to the
-  // updated autofill context.
-  [self cleanUpViewHierarchy:NO clearText:YES];
   [_activeView setTextInputClient:client];
   [_activeView reloadInputViews];
+
+  // Clean up views that no longer need to be in the view hierarchy, according to
+  // the current autofill context. The "garbage" input views are already made
+  // invisible to autofill and they can't `becomeFirstResponder`, we only remove
+  // them to free up resources and reduce the number of input views in the view
+  // hierarchy.
+  //
+  // This is scheduled on the runloop and delayed by 0.1s so we don't remove the
+  // text fields immediately (which seems to make the keyboard flicker).
+  // See: https://github.com/flutter/flutter/issues/64628.
+  [self performSelector:@selector(collectGarbageInputViews) withObject:nil afterDelay:0.1];
```

Comment:
@gaaclarke there must be a way to do this without a hard-coded wait right? I assume we just need to wait one frame? Does it fix it with just 0?

Reply:
I tried 0 it sometimes still flickers. 0.1 is a lot better on my test device.

Reply:
with 0 it flickers less but still does ~50% of the time. 0.1 works a lot better on my device.

Reply:
I wonder if if there are things in the new responder chain or whether there are focus setting imperative APIs we can call on the new field to avoid this race. 

What happens if you set this to 10s and type text in between or swipe away or cover the FlutterViewController in the meantime?

Reply:
Sorry had to fix my Xcode. I tried a longer delay and kept typing. Everything seemed to be fine.